doc/development/setup.md

```diff
@@ -0,0 +1,47 @@
+# Development Setup
+
+If you do not already have them you will need to:
+
+- [install west](https://docs.zephyrproject.org/3.6.0/develop/west/install.html)
+- [install east](https://github.com/IRNAS/irnas-east-software)
+
+Then follow these steps:
+
+```shell
+east init -m https://github.com/IRNAS/<repo-name> <repo-name>
+cd <repo-name>/project
+
+# Set up east globally (this only needs to be done once on each machine)
+east install nrfutil-toolchain-manager
+# Install toolchain for the version of NCS used in this project
+east install toolchain
+
+# Run `west update` via east to set up west modules in the repository
+east update
+
+# Create default VERSION files
+make gen-version
+```
+
+## Setup `pre-commit`
+
+Turn on `pre-commit` tool by running `pre-commit install`. If you do not have it yet **or the
+command did not succeed** follow instructions
+[here](https://github.com/IRNAS/irnas-guidelines-docs/tree/main/tools/pre-commit).
+
+## Building and flashing
+
+This repository contains several standalone firmware projects that can be build separately and are
+intended to be used for different use cases:
+
+- `app` - The main application firmware that will be used in the production. Found under `app/`
+  folder.
+- `sample` - Sample/example firmware, used either to demonstrate the use or functionality of smaller
+  parts of the main application firmware or used for some testing purpose, for example: hardware
+  bring-up tests. Samples are found under `samples/` folder.
+- `test` - Firmware used exclusively for unit testing testing or integration testing purposes. Tests
+  are found under `sample/` folder.
```

Comment:
I feel like this whole list of folder explanations should be somewhere else. In some of our repos, we have a "repository folder structure" section in the readme ([example](https://github.com/IRNAS/sentek-iotprobe-firmware?tab=readme-ov-file#repository-folder-structure)). 

I think maybe having something like that in the template as well would be good. Then all of this info can be in there.

Reply:
So this wasn't meant to be a folder explanation, but more of a explanation of what kinds of projects can be found in the repository (app, sample, test).

But you do make a good point, we could have a "repository folder structure" section in the top README.

I will create such section and remove/adapt the highlighted one.